### PR TITLE
feat: update heron's datasets

### PR DESCRIPTION
**Pull request by Ino-Ichan**
### Description

**This PR makes corrections and additions to the Dataset**
- Previously, the dataset was passing all tokens including padding as labels after tokenization, but padding should not be included in the loss calculation, so this has been corrected.
- When performing instruction tuning, it is correct to apply loss calculation only to the responses of the LLM, so a dataset that can calculate loss in this manner for the llava, m3it, and Japanese csv dataset has been added.

**Related Issue(s)**
- None

### Changes

- Fix: In the llava, m3it, and Japanese csv dataset, all padding tokens were being subjected to loss, so this has been changed by setting -100 to ensure they are ignored in loss calculations.
- Add: Prepared the llava, m3it, and Japanese csv instruct dataset for instruction tuning, which applies loss calculation only to the GPT responses.

Note: Setting a token to -100 will exclude it from loss and gradient calculations in the use of PyTorch's cross-entropy.

### Model/Algorithm Performance

- None

### Dependencies

- None

### Reviewer Notes

- The operation has been confirmed. Please point out if there are any areas of low readability.

### Confirmed

- [x] I have updated the documentation accordingly.
- [x] I have adhered to the coding standards and guidelines of this project.
- [x] I have added comments, especially in hard-to-understand areas.

### CODE REVIEW

File: heron/datasets/utils.py [hunk 1]
```diff
@@ -21,26 +21,55 @@
...
 
 from ..models.prepare_processors import get_processor
-from .ja_csv_datasets import JapaneseCSVDataset
-from .llava_datasets import LlavaDataset
-from .m3it_datasets import M3ITDataset
 
 
 def get_each_dataset(dataset_config: Dict, processor, max_length: int) -> Tuple[Dataset, Dataset]:
     if dataset_config["dataset_type"] == "m3it":
+        from .m3it_datasets import M3ITDataset
```
Comment (ymgaq) [inline heron/datasets/utils.py:28]: ここの処理ですが、動的importをしている点、train/validのdataset作成が基本的には繰り返しになっているのかなと思います。

もし遅延でimportする必要がないなら、例えば下記のようにしてはどうでしょう？これなら今後データセットクラスが拡張しても変更が容易かなと。

```
dataset_classes = {
    "m3it": M3ITDataset,
    "m3it_instruct": M3ITInstructDataset,
    "japanese_csv": JapaneseCSVDataset,
    "japanese_csv_instruct": JapaneseCSVInstructDataset,
    "llava": LlavaDataset,
    "llava_instruct": LlavaInstructDataset,
}

def get_each_dataset(dataset_config: Dict, processor, max_length: int) -> Tuple[Dataset, Dataset]:
    dataset_type = dataset_config["dataset_type"]
    if dataset_type not in dataset_classes:
        raise ValueError(f"dataset_type: {dataset_type} is not supported.")
    
    DatasetClass = dataset_classes[dataset_type]
    train_dataset = DatasetClass.create(dataset_config, processor, max_length, "train")
    val_dataset = DatasetClass.create(dataset_config, processor, max_length, "validation")

    return train_dataset, val_dataset
```

**Review by ymgaq — COMMENTED** [review]
何点かコメントしたので確認お願いします！

File: heron/datasets/ja_csv_datasets.py [hunk 2]
```diff
@@ -108,7 +109,14 @@ def _get_item_train(self, index):
         df_interest = self.loaded_dataset[self.loaded_dataset.img_path == img_path].reset_index(
             drop=True
         )
-        text = ""
+        # imageのロード
+        image = Image.open(os.path.join(self.dataset_root, img_path)).convert("RGB")
+        image = np.array(image)
+        if image.shape[2] != 3:
```
[the remark this answers is not in the document]
Comment (Ino-Ichan) [reply heron/datasets/ja_csv_datasets.py:115]: Redundantなので除きます！

File: heron/datasets/ja_csv_datasets.py [hunk 3]
```diff
@@ -91,8 +92,8 @@ def create(
     def __len__(self) -> int:
         return len(self.unique_img_path)
 
-    def preprocess_image(self, image):
-        return self.processor(images=[image], return_tensors="pt")["pixel_values"][0]
+    def preprocess_image(self, images):
+        return self.processor(images=images, return_tensors="pt")["pixel_values"][0]
```
[the remark this answers is not in the document]
Comment (Ino-Ichan) [reply heron/datasets/ja_csv_datasets.py:96]: 意図通りです！processorはバッチの軸が必要で、このような処理になってます。動画のときはまた別途考える必要があります。

File: heron/datasets/utils.py (hunk 1 above)
Comment (Ino-Ichan) [reply heron/datasets/utils.py]: 遅延importの必要はないです！こちらの実装でかなりスッキリするので反映しました！

File: configs/datasets/m3it_instruct.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+dataset_type: m3it
+dataset_names:
+  - textcap
+  - image-paragraph-captioning
+  - coco-goi
+  - coco-itm
```
Comment (kentosasaki-jp) [inline configs/datasets/m3it_instruct.yaml:6]: 今回のPRの本質ではないと思いますが、`dataset_names`の順序は特に指定ないですか？このくらい数がある場合は、アルファベット順になってると個人的には嬉しいです。

Comment (Ino-Ichan) [reply configs/datasets/m3it_instruct.yaml:6]: ここは[Hugging Faceのページ](https://huggingface.co/datasets/MMInstruction/M3IT#detailed-dataset-statistics)に記述されている順番になってます！

この指摘で気づいたのですが、上で[ymgさんが指摘してくれたdataset_classes](https://github.com/turingmotors/heron/pull/31#discussion_r1507351727)はアルファベット順になってたほうが良い気がしたので修正しました！

**Review by ymgaq — APPROVED** [review]
変更内容確認しました、LGTMです！

**Review by kentosasaki-jp — APPROVED** [review]
LGTM!
